src/common/film.c

```diff
@@ -200,6 +200,7 @@ int dt_film_new(dt_film_t *film, const char *directory)
                                 -1, &stmt, NULL);
     DT_DEBUG_SQLITE3_BIND_TEXT(stmt, 1, datetime, -1, SQLITE_STATIC);
     DT_DEBUG_SQLITE3_BIND_TEXT(stmt, 2, directory, -1, SQLITE_STATIC);
+    dt_pthread_mutex_lock(&darktable.db_insert);
```

Comment:
I'm wondering if the issue is now there. A lock has been removed.

Reply:
I'm not sure how this code can race, but sure, it's worth a try to bring back this lock. Though there are a lot of other places where DB is modified, but no lock is present.

Reply:
I didn't re-introdouce this lock as after analysis and testing it is not necessary to me.